tests/test_vec_envs.py

```diff
@@ -120,6 +120,68 @@ def make_env():
     vec_env.close()
 
 
+STEP_ENV_MAX_STEPS = 5
+
+
+class StepEnv(gym.Env):
+    def __init__(self):
+        """Gym environment for testing that terminal observation is inserted
+        correctly."""
+        self.action_space = gym.spaces.Discrete(2)
+        self.observation_space = gym.spaces.Box(np.array([0]), np.array([999]),
+                                                dtype='int')
+        self.max_steps = STEP_ENV_MAX_STEPS
+        self.current_step = 0
+
+    def reset(self):
+        self.current_step = 0
+        return np.array([self.current_step], dtype='int')
+
+    def step(self, action):
+        prev_step = self.current_step
+        self.current_step += 1
+        done = self.current_step >= self.max_steps
+        # return current step as observation
```

Comment:
I think this would be clearer without the comment (if you want to highlight which part is the observation, consider assigning it `obs = np.array(...)` and then `return obs, 0.0, done, {}`). If you want a comment then think need to unpack what current step is (and that it's *not* `self.current_step`!)